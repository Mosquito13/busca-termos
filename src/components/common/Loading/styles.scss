@import '../../../sass/styles.scss';

.loading {
    position: absolute;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    z-index: 99;
    overflow: hidden;
    display: flex;
    align-items: center;
    justify-content: center;
    transition: all .2s;
    opacity: 1;
    visibility: visible;
    @include themify() {
        background-color: themed('color-base');
    }

    &__icon {
        transform: translate(-10rem, -5rem);
    }

    &--loaded {
        opacity: 0;
        visibility: hidden;
    }
}