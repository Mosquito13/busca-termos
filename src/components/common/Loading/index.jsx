import PropTypes from 'prop-types';
import classNames from 'classnames';
import { PacmanLoader } from 'react-spinners';

import './styles.scss';

const Loading = ({ show }) => {
  const classes = classNames(
    'loading',
    !show && 'loading--loaded'
  );

  return (
    <div className={classes}>
      <div className="loading__icon">
        <PacmanLoader size={50} speedMultiplier={2} color="currentColor" />
      </div>
    </div>
  );
};

Loading.propTypes = {
  /**
   * Indica se exibe o carregando
   */
  show: PropTypes.bool
};

export default Loading;
