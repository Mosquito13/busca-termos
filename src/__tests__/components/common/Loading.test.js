import renderer from 'react-test-renderer';

import Loading from '../../../components/common/Loading';

describe(`${Loading.name}`, () => {
  describe('snapshots', () => {
    it('should render without show', () => {
      const tree = renderer.create(<Loading />).toJSON();

      expect(tree).toMatchSnapshot();
    });

    it('should render with show', () => {
      const tree = renderer.create(<Loading show />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
