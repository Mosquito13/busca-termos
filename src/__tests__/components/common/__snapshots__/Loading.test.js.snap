// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading snapshots should render with show 1`] = `
<div
  className="loading"
>
  <div
    className="loading__icon"
  >
    <span
      className="css-kf4vrf"
    >
      <span
        className="css-1u65263"
      />
      <span
        className="css-p9qef"
      />
      <span
        className="css-17ifzby"
      />
      <span
        className="css-1vh5u8m"
      />
      <span
        className="css-1up17uy"
      />
      <span
        className="css-lkr3bi"
      />
    </span>
  </div>
</div>
`;

exports[`Loading snapshots should render without show 1`] = `
<div
  className="loading loading--loaded"
>
  <div
    className="loading__icon"
  >
    <span
      className="css-kf4vrf"
    >
      <span
        className="css-1u65263"
      />
      <span
        className="css-p9qef"
      />
      <span
        className="css-17ifzby"
      />
      <span
        className="css-1vh5u8m"
      />
      <span
        className="css-1up17uy"
      />
      <span
        className="css-lkr3bi"
      />
    </span>
  </div>
</div>
`;
